### ISSUE TITLE
Could you provide an InnerJoin example?

### ISSUE THREAD

**Issue by alecava58** [mishael-o/Dapper.SimpleSqlBuilder#96]
### What happened?

Pls. provide a multiple InnerJoin example.

Thanks

### Reproduction steps

1.
2.
3.
...

### Version (bug)

latest

### Operating System

Windows

### Pull Request

- [ ] I would like to do a Pull Request

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lattaai3** [comment]
Hello, I tried to fix the issue

This is what I did:

This change adds a new test method to demonstrate multiple InnerJoin operations in the SelectBuilderTests class. The test showcases how to chain multiple InnerJoin calls and verifies the resulting SQL query.

I also created pull request: https://github.com/mishael-o/Dapper.SimpleSqlBuilder/pull/97


> [!CAUTION]
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!
>
> The fix provided by Latta AI might not be complete and it can serve as an inspiration.

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt fixing issues on your repository, you can block this account.

**Comment by mishael-o** [comment]
Hi @alecava58, I assume this is for the [select fluent builder](https://mishael-o.github.io/Dapper.SimpleSqlBuilder/docs/builders/fluent-builder/select-builder.html#join), if so then  you can do multiple joins this way:

```csharp
int minAge = 18;

var builder = SimpleBuilder.CreateFluent()
    .Select($"u.FirstName, u.LastName, u.Age, ut.Type, us.Status, ua.AddressLine1, ua.AddressLine2")
    .From($"User u")
    .InnerJoin($"UserType ut ON u.UserTypeId = ut.Id")
    .InnerJoin($"UserStatus us ON u.UserStatusId = us.Id")
    .InnerJoin($"UserAddress ua ON u.UserAddressId = ua.Id")
    .Where($"u.Age >= {minAge}");
``` 

This will produce this SQL statement.

```sql
SELECT u.FirstName, u.LastName, u.Age, ut.Type, us.Status, ua.AddressLine1, ua.AddressLine2
FROM User u
INNER JOIN UserType ut ON u.UserTypeId = ut.Id
INNER JOIN UserStatus us ON u.UserStatusId = us.Id
INNER JOIN UserAddress ua ON u.UserAddressId = ua.Id
WHERE u.Age >= @p0
```

You can check out the documentation for the [fluent builder](https://mishael-o.github.io/Dapper.SimpleSqlBuilder/docs/builders/fluent-builder/fluent-builder.html) for instructions on using other features.

---

However, if you are asking for the [builder](https://mishael-o.github.io/Dapper.SimpleSqlBuilder/docs/builders/builder.html) then can just write your dynamic SQL with your join statements.

**Comment by alecava58** [comment]
Fantastic.

Many thanks for the useful library.

Il Gio 7 Nov 2024, 20:18 Mishael Ogochukwu ***@***.***> ha
scritto:

**Comment by mishael-o** [comment]
Great! You are welcome.

I will be closing this issue. If you have any other issues to raise you are welcome to create a new issue.
